src/TorchVision/AutoAugment.cs

```diff
@@ -0,0 +1,172 @@
+            }
+
+            protected static Tensor apply_op(
+                                    Tensor img,
+                                    opType op_name,
+                                    float magnitude,
+                                    InterpolationMode interpolation,
+                                    IList<float>? fill)
+            {
+                switch(op_name) {
+                    case opType.ShearX:
+                        return F.affine(
+                            img,
+                            angle: 0.0f,
+                            translate: new[] { 0, 0 },
+                            scale: 1.0f,
+                            shear: new[] { 0.0f, (float)((180.0f / Math.PI) * Math.Atan(magnitude)) },
+                            interpolation: interpolation,
+                            fill: fill?.FirstOrDefault());
+                    case opType.TranslateX:
+                        return F.affine(
+                            img,
+                            angle: 0.0f,
+                            translate: new[] { (int)(magnitude), 0 },
+                            scale: 1.0f,
+                            interpolation: interpolation,
+                            shear: new[] { 0.0f, 0.0f },
+                            fill: fill?.FirstOrDefault());
+                    case opType.TranslateY:
+                        return F.affine(
+                            img,
+                            angle: 0.0f,
+                            translate: new[] { 0, (int)(magnitude) },
+                            scale: 1.0f,
+                            interpolation: interpolation,
+                            shear: new[] { 0.0f, 0.0f },
+                            fill: fill?.FirstOrDefault());
+                    case opType.Rotate:
+                        return F.rotate(img, magnitude, interpolation, fill: fill);
+                    case opType.Brightness:
+                        return F.adjust_brightness(img, 1.0 + magnitude);
+                    case opType.Color:
+                        return F.adjust_saturation(img, 1.0 + magnitude);
+                    case opType.Contrast:
+                        return F.adjust_contrast(img, 1.0 + magnitude);
+                    case opType.Sharpness:
+                        return F.adjust_sharpness(img, 1.0 + magnitude);
+                    case opType.Posterize:
+                        return F.posterize(img, (int)magnitude);
+                    case opType.Solarize:
+                        return F.solarize(img, magnitude);
+                    case opType.AutoContrast:
+                        return F.autocontrast(img);
+                    case opType.Equalize:
+                        return F.equalize(img);
+                    case opType.Invert:
+                        return F.invert(img);
+                    case opType.Identity:
+                        return img; // Pass
+                    default:
```

Comment:
I'm surprised the C# compiler doesn't complain about the fact that you have already covered all the possible cases. The body of the default case is, in effect, unreachable code.

Reply:
I take that back -- ShearY is not covered. Is that intentional?

Reply:
hmm, No, did it never have, or did I remove it in the refactoring I wonder.

Reply:
@NiklasGustafsson Good catch, I've fixed that now. 

BTW I think c# allows you to cast any int to an enum even when that enum doesn't contain that value, maybe that was an old quirk that would emit a warning now.